pkg/genericapiserver/routes/version.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package routes
+
+import (
+	"net/http"
+
+	"github.com/emicklei/go-restful"
+
+	"k8s.io/kubernetes/pkg/apiserver"
+	"k8s.io/kubernetes/pkg/version"
+)
+
+type Version struct{}
+
+// InstallVersionHandler registers the APIServer's `/version` handler
+func (v Version) Install(mux *apiserver.PathRecorderMux, c *restful.Container) {
```

Comment:
> The route setup can be generic, the actual version data isn't. Compare https://github.com/openshift/origin/blob/a58b6d87f531075ce9b9d3caffca66c00dfcab59/pkg/cmd/server/origin/master.go#L361-L361.

That's of dubious value.  Setting this endpoint for another server will break `kubectl` nearly 100% of the time and openshift wouldn't use this since we'd have different routes anyway.

I'm ok with the code for the route living in generic, but I don't think it should be called from the genericapiserver code.  Let the caller set it.